astropy/units/format/unicode_format.py

```diff
@@ -33,6 +35,23 @@ class Unicode(console.Console):
     def _format_mantissa(cls, m):
         return m.replace("-", "−")
 
+    @classmethod
+    def _format_unit_power(cls, unit, power=1):
+        superscript_unicode_units = [
+            u.arcsec,
+            u.arcmin,
+            u.degree,
+            u.hourangle,
+            u.electron,
+        ]
+        name = cls._get_unit_name(unit)
+        if power != 1 and unit in superscript_unicode_units:
```

Comment:
I'd follow the latex one a bit more closely, like:
```
name = cls._get_unit_name(unit)
if power != 1:
    if (short_name := unit.short_names[0]) in ("arcsec", "arcmin", "deg", "hourangle", "electron"):
        name  = short_name
    name += cls._format_superscript(utils.format_power(power))
return name
```
This has the advantage of not having to import `units` and making the comparison much faster (unit equality, which is implicit in your `unit in superscript_unicode_units` is much slower than string comparison).

p.s. Actually, it would be fine as well to just `if name == (..., "e⁻")`, i.e., compare with the unicode units we want to avoid...

Reply:
Yup, that makes sense.